clang/include/clang/Tooling/DependencyScanning/DependencyScanningWorker.h

```diff
@@ -130,11 +130,11 @@ class DependencyScanningWorker {
                                   DependencyActionController &Controller,
                                   StringRef ModuleName);
 
-  bool shouldEagerLoadModules() const { return EagerLoadModules; }
-
   llvm::vfs::FileSystem &getVFS() const { return *BaseFS; }
 
 private:
+  /// The parent dependency scanning service.
+  DependencyScanningService &Service;
```

Comment:
I'm guessing no one is likely to break this rule in practice, but I would add a note to the `DependencyScanningWorker` and `DependencyScanningTool` constructors that the service must outlive the worker.